### ISSUE TITLE
Unexpected results for same-length word set

### ISSUE THREAD

**Issue by tbrowder** [tbrowder/Abbreviations#1]
User Jim Bollinger sent the author an email reporting a result he was surprised by.  Example code replicating his results follows.

~~~
use Abbreviations;
my @list1 = <SAVE SEND SITE SOFT>;
my @list2 = <SAVE SITE SORT SORE>;

my %h1 = abbrev @list1, :out-type(AH);
my %h2 = abbrev @list2, :out-type(AH);
~~~

Said Jim: I get (as expected), the hash:

~~~
say %h1;
{
  SA   => "SAVE".Str,
  SAV  => "SAVE".Str,
  SAVE => "SAVE".Str,
  SE   => "SEND".Str,
  SEN  => "SEND".Str,
  SEND => "SEND".Str,
  SI   => "SITE".Str,
  SIT  => "SITE".Str,
  SITE => "SITE".Str,
  SO   => "SORT".Str,
  SOR  => "SORT".Str,
  SORT => "SORT".Str,
}
~~~

Jim said of the second list: But if I Abbrev the [second] list...I get (unexpectedly ), the hash:

~~~
say %h2;
{
  SAVE => "SAVE".Str,
  SEND => "SEND".Str,
  SITE => "SITE".Str,
  SORE => "SORE".Str,
  SORT => "SORT".Str,
}
~~~

Jim: What happened to the two & three letter abbreviations for the first three words?

**Comment by tbrowder** [comment]
I was surprised at first until I investigated the situation.  As I looked at my code I saw that I had already started working on the the published version for an update. The reason for the results is that the Rosetta Code subroutine is designed that way. It only calculates the minimum abbreviation length for the input set of words which in the case of the second list is four--hence the results as shown.  Whereas for the first list the calculated minimum abbreviation length is two resulting in what we intuitively expect.

~~~
# current version
my @list1 = <SAVE SEND SITE SOFT>; # minimum abbreviation length: 2
my @list2 = <SAVE SITE SORT SORE>; # minimum abbreviation length: 4
~~~

The algorithm for the next version will be changed to provide the results Jim, and I, expect:  maximum, unique abbreviations for a given word set.  In the algorithm for next release, each word in the input set will be treated individually to determine its complete abbreviation subset. Then, redundant abbreviations will be removed from the subset collection resulting in the largest possible unique abbreviation set for the input word list.

**Comment by tbrowder** [comment]
Closed by publishing version 2.0.0.
